library-sync.sh

```diff
@@ -3,6 +3,71 @@
 # utility script to gather template/imagestream content from https://github.com/openshift/library
 # and store it in this repo (cannot access other repos with dist git, and advised against git submodules
 
+# Without any commandline arguments, this script only updates the OCP supported samples. This behavior
+# can be modified using these commandline arguments:
+# * --okd - the OKD samples are updated
+# * --ocp - the OCP supported samples are updated
+# * --ocp-unsupported - all OCP samples are updated including the unsupported ones
+
+###########################################
+
+# process the commandline args
+PROCESS_OKD="false"
+PROCESS_OCP_UNSUPPORTED="false"
+if [[ $# -eq 0 ]]; then
+  PROCESS_OCP="true"
+else
+  PROCESS_OCP="false"
+fi
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+  --okd)
+    PROCESS_OKD="true"
+    ;;
+  --ocp)
+    PROCESS_OCP="true"
+    ;;
+  --ocp-unsupported)
+    PROCESS_OCP="true"
+    PROCESS_OCP_UNSUPPORTED="true"
+    ;;
+  esac
+  shift
+done
+
+# set up variables
+OCP_SUPPORTED_SAMPLES="ruby python nodejs perl php httpd nginx eap java webserver dotnet golang rails"
+OCP_ARCHS="ocp-x86_64 ocp-aarch64 ocp-ppc64le ocp-s390x"
+OKD_ARCHS="okd-x86_64"
+ALL_ARCHS="$OCP_ARCHS $OKD_ARCHS"
+if $PROCESS_OKD; then
+  if $PROCESS_OCP; then
+    SUPPORTED_ARCHS=$ALL_ARCHS
+  else
+    SUPPORTED_ARCHS=$OKD_ARCHS
+  fi
+else
+  SUPPORTED_ARCHS=$OCP_ARCHS
```

Comment:
it seems like by default it will synch all ocp archs , I mean if no flags are provided . Is this intentional ?

Reply:
I will rename the `SUPPORTED_ARCHS` to `PROCESSED_ARCHS` or something, because the "supported" is being used for 2 different things in the script which is confusing. There is a limited set of "supported ocp samples" and there's a list of architectures that is being processed in particular script run.

Reply:
Forgot to answer your actual question. Yes, by default the script will only process supported OCP samples in all OCP architectures. It leaves out OKD by default.